### PR TITLE
feat(prettier): add prettier & pre-commit hook for staged files for app and inside template

### PR DESCRIPTION
**Pull request by ruxandrafed**
Adding prettier & husky & linstaged to the actual NPM library starter kit, as well as inside the template, so that scaffolded libraries get it out of the box.

- add prettier
- add husky & set up pre-commit hook
- use lintstaged to auto-format only staged files

### CODE REVIEW

**Comment by billxinli** [comment]
🚢

File: .lintstagedrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  linters: {
+    "*.{js,jsx,json,yml,yaml,css,less,scss,md,graphql,mdx}": "npm run format",
```
Comment (maumercado) [inline .lintstagedrc]: 💡 : lintstaged files can take an array of commands, and so by running npm run format, a lot of files will be modified, and then the developer would have to git add them manually again, so my suggestion would be to modify this `npm run format` to `["npm run format", "git add"]`

Comment (ruxandrafed) [reply .lintstagedrc]: Great suggestion, added! https://github.com/telus/create-library/pull/52/commits/91763ac6a4b1fe6522ba76eb5aeca1c31589914c Thank you!

**Pull request merged**